packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,230 @@
+    const resHeaders: fetch.Headers = new fetch.Headers(
+      redisInfo.metadata.resHeaders
+    );
+    addCacheHeaders(
+      resHeaders,
+      "",
+      contentKey,
+      redisInfo.integrity,
+      redisInfo.time
+    );
+
+    // Return the response without a body for HEAD requests
+    if (req.method === "HEAD") {
+      return new fetch.Response(null, {
+        url: req.url,
+        headers: resHeaders,
+        status: 200
+      });
+    }
+
+    // Get the cached response body
+    const body: string = await this.keyv.get(getContentKey(req));
+
+    return Promise.resolve(
+      new fetch.Response(Buffer.from(body), {
+        url: req.url,
+        headers: resHeaders,
+        status: 200,
+        size: redisInfo.size
+      })
+    );
+  }
+
+  /**
+   * Takes both a request and its response and adds it to the given cache.
+   *
+   * @param req The request this is a storing a response for
+   * @param response The response to be stored
+   * @param opts
+   */
+  async put(
+    req: fetch.Request,
+    response: fetch.Response,
+    opts?: any
+  ): Promise<fetch.Response> {
+    opts = opts || {};
+    const size = response?.headers?.get("content-length");
+    const ckey = getContentKey(req);
+    const cacheOpts = {
+      algorithms: opts.algorithms,
+      integrity: null,
+      metadata: {
+        url: req.url,
+        reqHeaders: req.headers.raw(),
+        resHeaders: response.headers.raw()
+      },
+      size,
+      time: Date.now()
+    };
+    if (req.method === "HEAD" || response.status === 304) {
+      // Update metadata without writing
+      const redisInfo = await this.keyv.get(getMetadataKey(req));
+      // Providing these will bypass content write
+      cacheOpts.integrity = redisInfo.integrity;
+      addCacheHeaders(
+        response.headers,
+        "",
+        ckey,
+        redisInfo.integrity,
+        redisInfo.time
+      );
+      await this.keyv.set(getMetadataKey(req), cacheOpts);
+
+      return response;
+    }
+
+    const body = await response.text();
+
+    await this.keyv.set(getMetadataKey(req), cacheOpts);
+
+    await this.keyv.set(getContentKey(req), body);
+
+    return Promise.resolve(new fetch.Response(Buffer.from(body), response));
+  }
+
+  /**
+   * Finds the Cache entry whose key is the request, and if found, deletes the
+   * Cache entry and returns a Promise that resolves to true. If no Cache entry
+   * is found, it returns false.
+   */
+  async delete(req: fetch.Request, opts?: any): Promise<boolean> {
+    return (
+      (await this.keyv.delete(getMetadataKey(req))) ||
+      (await this.keyv.delete(getContentKey(req)))
+    );
+  }
+
+  /**
+   * Redis will attempt to maintain an active connection which prevents Node.js
+   * from exiting. Call this to gracefully close the connection.
+   */
+  async quit(): Promise<boolean> {
+    return (await this.keyv?.opts?.store?.redis?.quit()) == "OK";
```

Comment:
Don't we need strict check for "OK" using "==="?

Reply:
Fixed